File: erp-frontend/src/api/files.ts
import client from './client'

export interface PresignedUrlPayload {
  filename: string
  content_type: string
  folder?: string
}

export interface PresignedUrlResult {
  upload_url: string
  file_key: string
  file_url: string
}

export const filesApi = {
  async createPresignedUrl(payload: PresignedUrlPayload): Promise<PresignedUrlResult> {
    const response = await client.post<PresignedUrlResult>('/files/presigned-url', payload)
    return response.data
  },

  async deleteObject(objectKey: string): Promise<void> {
    await client.delete('/files/object', {
      params: {
        object_key: objectKey,
      },
    })
  },
}

File: erp-backend/app/services/files.py
from __future__ import annotations

from app.core.storage import create_presigned_upload
from app.schemas.file import PresignedUrlRequest, PresignedUrlResponse


class FileService:
    async def create_presigned_url(self, data: PresignedUrlRequest) -> PresignedUrlResponse:
        upload_url, file_key, file_url = await create_presigned_upload(
            filename=data.filename,
            content_type=data.content_type,
            folder=data.folder,
        )
        return PresignedUrlResponse(
            upload_url=upload_url,
            file_key=file_key,
            file_url=file_url,
        )
